tests/filewriter_test.cpp

```diff
@@ -0,0 +1,66 @@
+/**
+ * @file filewriter_test.cpp
+ */
+#include "catch.hpp"
+#include "mlboard.hpp"
+#include <unistd.h>
+#include <sstream>
+#include <cstdio>
+#include <sys/stat.h>
+/**
+ * Test the 2 filewriter objects at a time
+ */
+TEST_CASE("Writing two files at a time", "FileWritter")
+{
+    // Both files created at same time and hence there should be only one file
+    mlboard::FileWriter f1,f2;
```

Comment:
I haven't seen it fail currently, may be once merge and it will run with all pr and commits and then we see if it fails often we can remove it or debug it

Reply:
Okay, fine :+1:

Reply:
```suggestion
    mlboard::FileWriter f1, f2;
```